common/djangoapps/enrollment/views.py

```diff
@@ -783,3 +785,82 @@ def post(self, request):
                     actual_activation=current_enrollment['is_active'] if current_enrollment else None,
                     user_id=user.id
                 )
+
+
+@can_disable_rate_limit
+class EdraakCourseListView(APIView, ApiKeyPermissionMixIn):
+    """
+    """
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser,
+                              EnrollmentCrossDomainSessionAuth,)
+    permission_classes = ApiKeyHeaderPermissionIsAuthenticated,
+    throttle_classes = EnrollmentUserThrottle,
+
+    # Since the course about page on the marketing site
+    # uses this API to auto-enroll users, we need to support
+    # cross-domain CSRF.
+    @method_decorator(ensure_csrf_cookie_cross_domain)
+    def get(self, request):
+        """
+        GET for endpoint api/enrollment/v1/edraak_course_list
+        Accepts the following optional parameter:
+            is_certificate_allowed
+            is_certificate_available
+            is_completed
+
+        The API will use get_course_enrollments and filter-out records that do no match the criteria given in
+        optional parameters list.
+
+        Example: api/enrollment/v1/edraak_course_list?is_completed=true
+            get all enrollments for current user, where courses are completed, regardless of certificate status
+        """
+        try:
+            enrollment_data = get_course_enrollments(
+                request.user.username
+            )
+        except CourseEnrollmentError:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": (
+                        u"An error occurred while retrieving enrollments for user '{username}'"
+                    ).format(username=request.user.username)
+                }
+            )
+
+        try:
+            _certificate_allowed = self._get_boolean(request=request, param='is_certificate_allowed')
+            _certificate_available = self._get_boolean(request=request, param='is_certificate_available')
+            _completed = self._get_boolean(request=request, param='is_completed')
+        except ValueError as error:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"message": str(error)}
+            )
+
+        final_data = []
+        for enrollment in enrollment_data:
+            if (
+                    self._filter_out(_certificate_allowed, six.text_type(enrollment['is_certificate_allowed'])) or
+                    self._filter_out(_certificate_available, six.text_type(enrollment['is_certificate_available'])) or
+                    self._filter_out(_completed, six.text_type(enrollment['is_completed']))
+            ):
+                continue
+            final_data.append(enrollment)
```

Comment:
Rather than complicating space requirement here, why not go for pythonic syntax:
`enrollment_data = [enrollment for enrollment in enrollment_data if _filtered_enrollment(params, enrollment)]` where `_filtered_enrollment` does the logic related to checking if an enrollment meets the set filters from params. What do you think?